File: dotnet/DesignPatterns/CreationalDesignPatterns/PrototypeDesignPattern.Example3/Rectangle.cs
namespace PrototypeDesignPattern.Example3;

public class Rectangle : Shape
{
	public double Width { get; set; }
	public double Height { get; set; }

	public Rectangle(string color, double x, double y, double width, double height) : base(color, x, y)
	{
		Width = width;
		Height = height;
	}

	private Rectangle(Rectangle other) : base(other.Color, other.X, other.Y)
	{
		Width = other.Width;
		Height = other.Height;
	}

	public override object Clone()
	{
		return new Rectangle(this);
	}

	public override void Display()
	{
		Console.WriteLine($"Rectangle at ({X}, {Y}) with dimensions {Width}x{Height} and color {Color}");
	}
}

File: java/DesignPatterns/CreationalDesignPatterns/PrototypeDesignPattern/PrototypeDesignPattern.Example3/src/main/java/creational/prototype/example3/Rectangle.java
package creational.prototype.example3;

public class Rectangle extends Shape {
    public double width;
    public double height;

    public Rectangle(String color, double x, double y, double width, double height) {
        super(color, x, y);
        this.width  = width;
        this.height = height;
    }

    private Rectangle(Rectangle other) {
        super(other.color, other.x, other.y);
        this.width  = other.width;
        this.height = other.height;
    }

    @Override
    public Object clone() { return new Rectangle(this); }

    @Override
    public void display() {
        System.out.println("Rectangle at (" + x + ", " + y + ") with dimensions " + width + "x" + height + " and color " + color);
    }
}
